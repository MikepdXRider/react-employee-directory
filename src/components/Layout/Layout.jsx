import React from 'react'
import { useHistory } from 'react-router-dom';
import useAuth from '../../hooks/useAuth.jsx';

export default function Layout({ children }) {
    const history = useHistory();
    
    const { user, signOut } = useAuth();
    
    async function handleLogout() {
        await signOut();
    }

    return (
        <>
            <header>
                <h1>Acme Inc</h1>
                {
                    user.email
                    ? <>
                        <span>You are signed in as {user.email}</span>
                        <button onClick={() => handleLogout()}>Sign Out</button> 
                    </> : <button onClick={() => history.push('/login')}>Sign In</button>
                }
            </header>
            <main>
                {children}
            </main>
        </>
    )
}
