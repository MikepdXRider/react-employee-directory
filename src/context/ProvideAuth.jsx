// 
// UserContext
// import hook and createContext
// call createContext hook
// create UserProvider w/ initialized state
//  - pass state mutators and any relevant functions into provider value attribute
//  - export provider and context
// 