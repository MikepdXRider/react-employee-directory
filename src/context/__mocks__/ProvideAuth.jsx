import { createContext, useState } from "react";

const authContext = createContext();

function ProvideAuth({mockUser, children}) {
    // on app load...
    // getUser checks the 'session' in localstorage for a current token and a refresh token. If the token is current, it looks for a user in the session. If the token is not current and a refresh token is present, it refreshes the token and retrieves a new user.
    const [user, setUser] = useState(
        mockUser ? {email: mockUser.email} : {}
    );

    return (
        <authContext.Provider value={{user}} >
            {children}
        </authContext.Provider>
    )
}

export { authContext, ProvideAuth}