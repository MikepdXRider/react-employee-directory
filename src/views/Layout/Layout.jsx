// 
// Layout View Component
// Needs access to auth/user context
// Contains header
//  - Header contains:
//      - Acme Inc. name
//      - A “Sign In” button if the user is signed out
//      - A “You are signed in as ${email}”` message with a “Sign Out” button if the user is signed in
// 


import React from 'react'

export default function Layout({ children }) {
    return (
        <>
            <header>
                <h1>Acme Inc</h1>
                {/* “Sign In” button or “You are signed in as ${email}” message with a “Sign Out” button */}
            </header>
            <main>
                {children}
            </main>
        </>
    )
}
