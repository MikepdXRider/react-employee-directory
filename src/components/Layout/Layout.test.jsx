import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ProvideAuth } from '../../context/ProvideAuth.jsx'
import Layout from './Layout.jsx'

// jest.mock('../../services/users.js');
jest.mock('../../context/ProvideAuth.jsx');

it('renders layout w/o user', () => {
    const {container} = render(
        <ProvideAuth>
            <MemoryRouter>
                <Layout />
            </MemoryRouter>
        </ProvideAuth>
    )

    expect(container).toMatchSnapshot();
})

it('renders layout w/ user', () => {
    const {container} = render(
        <ProvideAuth mockUser={{email: 'hi@test.com'}}>
            <MemoryRouter>
                <Layout />
            </MemoryRouter>
        </ProvideAuth>
    )

    expect(container).toMatchSnapshot();
})