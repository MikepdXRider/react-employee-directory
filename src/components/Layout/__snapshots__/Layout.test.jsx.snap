// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders layout 1`] = `
<div>
  <header>
    <h1>
      Acme Inc
    </h1>
    <button>
      Sign In
    </button>
  </header>
  <main />
</div>
`;
